Add architectural information to database guides

ptgott opened the pull request:
Closes #21540

Rather than attempt to describe the entirety of the database access architecture in the `/docs/database-access/architecture/` guide, this change:

- Edits the architecture guide to clarify that authentication between the Teleport Database Service and each proxied database depends on the database.
- Adds a "How it works" H2 before the "Prerequisites" H2 of each how-to guide for accessing a specific database, describing the architecture of the setup shown in the guide.

ptgott committed:
Add architectural information to database guides

Closes #21540

Rather than attempt to describe the entirety of the database access
architecture in the `/docs/database-access/architecture/` guide, this
change:

- Edits the architecture guide to clarify that authentication between
  the Teleport Database Service and each proxied database depends on the
  database.
- Adds a "How it works" H2 before the "Prerequisites" H2 of each how-to
  guide for accessing a specific database, describing the architecture
  of the setup shown in the guide.

greedy52 commented on docs/pages/database-access/guides/redis-aws.mdx:
```diff
@@ -5,6 +5,11 @@ description: How to configure Teleport database access with AWS ElastiCache and
 
+## How it works
+
+(!docs/pages/includes/database-access/how-it-works/iam.mdx db="ElastiCache or
+MemoryDB for Redis cluster" cloud="AWS"!)
```
We have two modes for AWS redis
- IAM auth for Redis 7.0+
- Teleport-managed users using Secrets manager

Details see https://goteleport.com/docs/database-access/guides/redis-aws/#step-56-configure-authentication-for-elasticache-or-memorydb-users

greedy52 reviewed (commented):
Thanks Paul! this is awesome!

ptgott committed:
Respond to greedy52 feedback

- Correct the DynamoDB and OpenSearch architecture descriptions.
- Correct the AWS Redis architecture description by moving a description
  from later in the page. Also adds Redis ACL to the prerequisites.
- Edit the Azure Cache for Redis and Azure MySQL and PostgreSQL guides
  to say "Microsoft Entra ID" in the architecture section. This change
  does not attempt to correct "Azure AD" elsewhere in the guides, as
  this entails fixing UI instructions that are outside the scope of the
  PR.

ptgott commented:
@r0mant Would you have time to give this one a look? Thanks!

r0mant commented on docs/pages/database-access/guides/aws-opensearch.mdx:
```diff
@@ -5,6 +5,16 @@ description: How to access AWS OpenSearch with Teleport database access
+which maps an IAM role to a backend OpenSearch role. Teleport users choose an
+IAM role and connect to OpenSearch via a local proxy server. The local proxy
+server forwards requests to the Teleport Database Service via the Teleport Proxy
+Service.The Database Service assumes the user's chosen IAM role, rewrites
```
```suggestion
Service. The Database Service assumes the user's chosen IAM role, rewrites
```

r0mant commented on docs/pages/database-access/guides/redis-aws.mdx:
```diff
@@ -5,6 +5,22 @@ description: How to configure Teleport database access with AWS ElastiCache and
+  the database using a short-lived AWS IAM authentication token. AWS IAM
+  authentication is available for ElastiCache and MemoryDB with Redis version
+  7.0 or above.
+- **Managing users:** The Teleport Database Service manages users in a Redis
+  access control list, rotates their passwords every 15 minutes, and saves these
+  passwords in AWS Secrets Manager. The Database Service automatically sends an
+  `AUTH` command with the saved password when connecting the client to the Redis
+  server.
```
@greedy52 Think we can deprecate this method anytime soon?

greedy52 replied:
IAM auth is only supported for Redis 7.0 +, so it will take a while to deprecate the old method:
https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/supported-engine-versions.html

r0mant reviewed (approved):
:shipit:
